src/dict.c

```diff
@@ -110,6 +111,16 @@ dict *dictCreate(dictType *type)
     return d;
 }
 
+/* Create an array of dictionaries */
+dict **dictCreateMultiple(dictType *type, int count)
+{
+    dict **d = zmalloc(sizeof(dict*) * count);
+    for (int i = 0; i < count; i++) {
+        d[i] = dictCreate(type);
+    }
```

Comment:
I've initially done it as an array, but after discussion with @madolson we've decided that dict** is cleaner as it provides better static error checking. With dict*, there is a chance you could accidentally pass an array where specific dict is expected and dict[0] will be used instead of the slot specific dict. In fact I've done this error couple times myself and corrected it only after changing the type to dict**.

Reply:
I don't think there is a performance hit, given that array size is relatively large it wouldn't be pre-fetched, so we'll probably end up with extra memory reads anyways. From brief benchmarking that I've done, CRC hash sum calculation (from `getDict`) was the only clear place where we've introduced some slowdown. Once we eliminate that, I'm going to run more thorough benchmarks, and we can discuss results and see if anything else is an issue. Do you agree with this plan?

Reply:
Good plan. Measuring is what matters.

Theoretically though (and since thinking about it is more fun than actually measuring) I believe accessing some field in the dict, let's say rehashidx, in `dict_pointers[slot]->rehashidx` it is two reads (one to fetch the pointer and one to follow it and fetch the dict struct) while in `dict_array[slot].rehashidx` it is only one read.

Reply:
Yes, but remember not all pointers are created equal. The multiple pointer bit is important for normal keys and values because they probably aren't in any of the CPU caches, so you have to pay a full DRAM miss, whereas there is a *good* chance that some of these dictionaries will be in cache. Still worth doing performance checking though.

Reply:
I don't see any measurable performance degradation from this, my assumption is that because in non-cluster mode there is only one pointer per DB, it can stay in CPU cache.